File: frontend/src/api/app-schedule.ts
import request from './request'
import type { PageResult } from './types'

/** 定时任务 */
export interface AppSchedule {
  id: number
  tenantId: number
  name: string
  appId: number
  appName?: string
  /** interval间隔 / daily每天 / weekly每周 */
  triggerType: string
  /** 间隔分钟 */
  intervalMinutes?: number
  /** HH:mm */
  runTime?: string
  /** 1-7 周一~周日 */
  runWeekday?: number
  /** 触发输入 */
  inputMessage?: string
  enabled: number
  lastRunTime?: string
  nextRunTime?: string
  remark?: string
  createTime?: string
  updateTime?: string
}

/** 执行记录 */
export interface AppScheduleLog {
  id: number
  scheduleId: number
  scheduleName?: string
  appId?: number
  appName?: string
  /** scheduled/manual */
  triggerBy: string
  /** success/failed */
  status: string
  message?: string
  costMs?: number
  createTime?: string
}

export interface SchedulePayload {
  name: string
  appId?: number
  triggerType?: string
  intervalMinutes?: number
  runTime?: string
  runWeekday?: number
  inputMessage?: string
  enabled?: number
  remark?: string
}

export interface RunOutcome {
  status: string
  message: string
  costMs: string
}

/** 定时任务 API（/api/app/schedules） */
export const scheduleApi = {
  page(params: { page?: number; size?: number; keyword?: string; enabled?: number }) {
    return request.get<never, PageResult<AppSchedule>>('/app/schedules', { params })
  },
  get(id: number) {
    return request.get<never, AppSchedule>(`/app/schedules/${id}`)
  },
  create(data: SchedulePayload) {
    return request.post<never, AppSchedule>('/app/schedules', data)
  },
  update(id: number, data: SchedulePayload) {
    return request.put<never, void>(`/app/schedules/${id}`, data)
  },
  setEnabled(id: number, enabled: boolean) {
    return request.put<never, void>(`/app/schedules/${id}/enabled`, null, { params: { enabled } })
  },
  run(id: number) {
    return request.post<never, RunOutcome>(`/app/schedules/${id}/run`)
  },
  remove(id: number) {
    return request.delete<never, void>(`/app/schedules/${id}`)
  },
  logs(id: number, params: { page?: number; size?: number }) {
    return request.get<never, PageResult<AppScheduleLog>>(`/app/schedules/${id}/logs`, { params })
  }
}

File: agent-dao/src/main/java/com/agent/platform/dao/entity/app/AppSchedule.java
package com.agent.platform.dao.entity.app;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 应用定时任务配置
 */
@Data
@TableName("app_schedule")
public class AppSchedule {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long tenantId;

    /** 任务名称 */
    private String name;

    /** 关联应用ID */
    private Long appId;

    /** 应用名称快照 */
    private String appName;

    /** 触发类型：interval间隔 / daily每天 / weekly每周 */
    private String triggerType;

    /** 触发间隔(分钟)，interval 生效 */
    private Integer intervalMinutes;

    /** 触发时刻 HH:mm，daily/weekly 生效 */
    private String runTime;

    /** 周几 1-7（周一~周日），weekly 生效 */
    private Integer runWeekday;

    /** 触发时发送给应用的输入 */
    private String inputMessage;

    /** 是否启用：0停用 1启用 */
    private Integer enabled;

    /** 最近执行时间 */
    private LocalDateTime lastRunTime;

    /** 下次执行时间 */
    private LocalDateTime nextRunTime;

    /** 备注 */
    private String remark;

    /** 创建人 */
    private Long createBy;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
}
